var/spack/repos/builtin/packages/py-nrm/package.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyNrm(PythonPackage):
+    """Python interface to the Argo Node Resource Manager,
+    an infrastructure to allow applications to manage node resources
+    on the fly."""
+
+    homepage = "https://nrm.readthedocs.io"
+
+    maintainers = ['perarnau']
+
+    version('main', url='https://github.com/anlsys/nrm-python/archive/refs/heads/master.zip')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('nrm-core@0.7.0')
+    depends_on('py-pyzmq', type=('build', 'run'))
+    depends_on('py-tornado', type=('build', 'run'))
+    depends_on('py-jsonschema', type=('build', 'run'))
+
```

Comment:
there's stuff missing in there, and I think the versions might be critical:

- warlock (python package) 1.3.3
- tornado v5.1
- jsonschema v 3.2.0
- msgpack (python)
- linux perf (not sure if available in spack)

Reply:
Unfortunately I couldn't see anything resembling a perf package in spack. Confusingly many packages use "perf" to refer to post-build performance or benchmark checks.

Reply:
Doesn't look like it's in spack, which might be an issue...I'll have to fix it by migrating it to C & PAPI inside nrm-extra. Let's forget about that for now.